src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIClient.java

```diff
@@ -117,4 +128,15 @@ protected String generateAccessToken(String restApiEndpoint, String clientId, St
     token = client.accessToken(request, OAuth.HttpMethod.POST, OAuthJSONAccessTokenResponse.class).getAccessToken();
     return token;
   }
+
+  private void validateHttpResponse(HttpResponse response) throws IOException {
+    int code = response.getStatusLine().getStatusCode();
+    if (RETRYABLE_CODES.contains(code)) {
+      throw new RetryableException(String.format(HTTP_ERROR_MESSAGE, code));
+    } else if (SUCCESS_CODES.contains(code)) {
```

Comment:
again no-op piece of code. You could do !SUCCESS_CODES.contains(code) throw new IOException....

Reply:
Also as discussed, this should ideally throw some other exception no IOException as it is mainly parsing the response.

Reply:
It's better to return early to simply the structure.

```
if (SUCCESS_CODES.contains(code)) {
  return;
}
if (RETRYABLE_CODES.contains(code)) {
  throw new RetryableException(....);
}
throw new IOException(...);
```